modules/Core/pages/login.php

```diff
@@ -29,44 +29,22 @@
 $method = $queries->getWhere('settings', array('name', '=', 'login_method'));
 $method = $method[0]->value;
 
-// Use recaptcha?
-$recaptcha = $queries->getWhere('settings', array('name', '=', 'recaptcha_login'));
-$recaptcha = count($recaptcha) ? $recaptcha[0]->value : 'false';
-
-$captcha_type = $queries->getWhere('settings', array('name', '=', 'recaptcha_type'));
-$captcha_type = $captcha_type[0]->value;
-
-$recaptcha_key = $queries->getWhere("settings", array("name", "=", "recaptcha_key"));
-$recaptcha_secret = $queries->getWhere('settings', array('name', '=', 'recaptcha_secret'));
+// Use captcha?
+$captcha = $queries->getWhere('settings', array('name', '=', 'recaptcha_login'));
+$captcha = count($captcha) ? $captcha[0]->value : 'false';
 
 // Deal with input
 if (Input::exists()) {
 	// Check form token
 	if (Token::check()) {
 		// Valid token
-		if (!isset($_SESSION['tfa']) && $recaptcha == 'true') {
-			// Check captcha
-			$url = $captcha_type === 'hCaptcha' ? 'https://hcaptcha.com/siteverify' : 'https://www.google.com/recaptcha/api/siteverify';
-
-			$post_data = 'secret=' . $recaptcha_secret[0]->value . '&response=' . ($captcha_type === 'hCaptcha' ? Input::get('h-captcha-response') : Input::get('g-recaptcha-response'));
-
-			$ch = curl_init($url);
-			curl_setopt($ch, CURLOPT_POST, 1);
-			curl_setopt($ch, CURLOPT_POSTFIELDS, $post_data);
-			curl_setopt($ch, CURLOPT_HEADER, 0);
-			curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-
-			$result = curl_exec($ch);
-
-			$result = json_decode($result, true);
+		if (!isset($_SESSION['tfa']) && $captcha == 'true') {
+            $captcha_passed = CaptchaBase::getActiveProvider()->validateToken($_POST);
```

Comment:
Not sure if this is just GH preview, but line 42 appears to be not indented properly.

Reply:
it's because the file contains tabs, which many editors display as 4 spaces but github displays as 8

Reply:
For now I'll change line 42 to tabs too, until we run a formatter through the whole project to convert to spaces